applications/GeoMechanicsApplication/custom_processes/apply_k0_procedure_process.hpp

```diff
@@ -0,0 +1,161 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//
+
+#pragma once
+
+// System includes
+#include <cmath>
+#include <iostream>
+#include <string>
+
+// Project includes
+#include "includes/model_part.h"
+#include "utilities/openmp_utils.h"
+#include "utilities/math_utils.h"
+#include "includes/element.h"
+
+// Application includes
+
+#include "includes/kratos_flags.h"
+#include "includes/kratos_parameters.h"
+#include "processes/process.h"
+
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+class ApplyK0ProcedureProcess : public Process
+{
+  public:
+
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyK0ProcedureProcess);
+
+    //----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    /// Constructor
+    ApplyK0ProcedureProcess(ModelPart&  model_part,
+                           Parameters& rParameters) : Process(Flags()), mrModelPart(model_part)
+    {
+        KRATOS_TRY
+
+        mModelPartName      =  rParameters["model_part_name"].GetString();
+
+        KRATOS_CATCH("")
+    }
+
+    ///------------------------------------------------------------------------------------
+
+    /// Destructor
+    ~ApplyK0ProcedureProcess() override{}
+
+    //----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    void ExecuteFinalizeSolutionStep() override
+    {
+        KRATOS_TRY
+
+        if (mrModelPart.NumberOfElements() > 0) {
+             // K0 procedure for the model part:
+             block_for_each(mrModelPart.Elements(), [this](Element& rElement) {
+                 CalculateK0Stresses(rElement);
+              });
+         }
+ 
+        KRATOS_CATCH("")
+    }
+
+    //----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+   
+  protected:
+    /// Member Variables
+
+    ModelPart& mrModelPart;
+    std::string mModelPartName;
+
+  private:
+
+      void CalculateK0Stresses(Element& rElement)
+      {
+          // Get K0 material parameters of this element ( probably there is something more efficient )
+          const Element::PropertiesType& rProp = rElement.GetProperties();
+          ConstitutiveLaw::Pointer pConstitutiveLaw = rProp.GetValue(CONSTITUTIVE_LAW);
+          const int& K0MainDirection = rProp[K0_MAIN_DIRECTION];
```

Comment:
I would change this reference-to-const to a const-value, since this is about a built-in type (`int`).
```suggestion
          const int K0MainDirection = rProp[K0_MAIN_DIRECTION];
```

Reply:
Done